src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -0,0 +1,337 @@
+package org.broadinstitute.hellbender.cmdline.GATKPlugin;
+
+import com.sun.jersey.server.impl.wadl.WadlResource;
+import htsjdk.samtools.SAMFileHeader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * A GATKCommandLinePluginDescriptor for ReadFilter plugins
+ */
+public class GATKReadFilterPluginDescriptor extends GATKCommandLinePluginDescriptor<ReadFilter> {
+
+    protected final Logger logger = LogManager.getLogger(this.getClass());
+
+    private static final String pluginPackageName = "org.broadinstitute.hellbender.engine.filters";
+    private static final Class<?> pluginBaseClass = org.broadinstitute.hellbender.engine.filters.ReadFilter.class;
+
+    @Argument(fullName = StandardArgumentDefinitions.READ_FILTER_LONG_NAME,
+            shortName = StandardArgumentDefinitions.READ_FILTER_SHORT_NAME,
+            doc="Read filters to be applied before analysis", optional=true)
+    public final List<String> userReadFilterNames = new ArrayList<>(); // preserve order
+
+    final static String disableReadFilterArgName = "disableReadFilter";
+    @Argument(fullName = disableReadFilterArgName,
+            shortName = "DF",
```

Comment:
Sorry for the meddling, but while trying to understand how a plugin will be implemented I found this: the names for the read filter are in `StandardArgumentDefinitions`, the disable read filter in a static field (only the long name, the other is hardcoded in the annotation), and the one for disable all the read filters is inside the annotation. I think that it would be nice to have all this values in `StandardArgumentDefinitions` to be able to have a class to check possible collisions...